torch/nn/functional.py

```diff
@@ -547,6 +547,22 @@ def nll_loss(input, target, weight=None, size_average=True):
     return f(input, target)
 
 
+def poisson_nll_loss(log_input, target, size_average=True):
+    r"""Poisson negative log likelihood loss.
+
+    Drops :math:`log(target!)` term. See :class:`~torch.nn.PoissonNLLLoss` for details.
+
+    Args:
+        log_input: :math:`log(input)`
+        target: :math:`target \sim Pois(input)`
+    """
+    loss = torch.exp(log_input) - target * log_input
```

Comment:
I see.
This loss seems a bit too niche, but as it's implemented in TF then maybe it would make sense to have it.
Maybe it would be good then to also add the stirling's approximation for the loss as an argument? It seems that we can easily implement it following [TF implementation](https://github.com/tensorflow/tensorflow/blob/r1.2/tensorflow/python/ops/nn_impl.py#L88-L96)?